### PR TITLE
Hide handouts pdf if there are no handouts in it

### PR DESCRIPTION
**Pull request by bethanyaconnor**
We got a report that the "Print Handouts" button on the new HOC lessons showed a blank PDF, which makes sense as there aren't any handouts. In this PR, we will pass down a nil URL for the handouts pdf, which the frontend already deals with correctly (I also added a test).

I checked:
- /s/hello-world-food-2021/lessons/1 and /s/hello-world-food-2021 no longer have a handout link
- /s/coursea-2021/lessons/1 and /s/coursea-2021/ still have a handout link

Let me know if there are any other cases to check!

Also spent this opportunity to deduplicate some code.

## Testing story

unit tests

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/models/script_test.rb [hunk 1]
```diff
@@ -931,6 +932,16 @@ def populate_cache_and_disconnect_db
     assert_equal '/resources-pdf-url', summary[:scriptResourcesPdfUrl]
   end
 
+  test 'get_unit_resources_pdf_url returns nil if no resources in script or lessons' do
+    Services::CurriculumPdfs.stubs(:get_unit_resources_url).returns('/resources-pdf-url')
+    unit = create(:script, name: 'single-lesson-script', instruction_type: SharedCourseConstants::INSTRUCTION_TYPE.teacher_led, instructor_audience: SharedCourseConstants::INSTRUCTOR_AUDIENCE.teacher, participant_audience: SharedCourseConstants::PARTICIPANT_AUDIENCE.student)
```
Comment (Erin007) [inline dashboard/test/models/script_test.rb]: Certainly doesn't need to be part of this PR, but it is on EPT's radar to create script factories with the new instruction types? Kind of like how we have ["shortcut" factories](https://github.com/code-dot-org/code-dot-org/blob/0ca01460d647b30a70f583f5930f7e790b5f3407/dashboard/test/factories/factories.rb#L779) to make scripts of different CSX types? Could be one way to keep tests a tad tidier and use factories as sort of documentation for the different script types.

**Review by Erin007 — APPROVED** [review]
Thanks for the extra clean up and tests. LGTM!

File: dashboard/test/models/script_test.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/test/models/script_test.rb]: That's a fair question but I did just copy this test without reading too closely and it turns out I don't need this extra stuff 🙈 @dmcavoy has been working in this area so she would have the most context on whether we should add these kinds of factories here

Comment (davidsbailey) [reply dashboard/test/models/script_test.rb]: great call @Erin007 ! +1 to setting up more factory shortcuts. we have not been great about this in our area in general, for example I just added a test with a bunch of copy-pasted lines just to have a script containing a level: https://github.com/code-dot-org/code-dot-org/blob/f75b1b6db38882874b181c051e3062a0418583ec/dashboard/test/controllers/scripts_controller_test.rb#L1654-L1660

**Pull request merged**